tests/unit/schema/schemaManager/schemaManger.spec.ts

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { SchemaManager } from '../../../../src/schema/models/schemaManager';
+
+let schemaManager: SchemaManager;
+
+describe('SchemaManager', () => {
+  let getSchema: jest.Mock;
+  let getSchemas: jest.Mock;
+
+  const schema = {
+    name: 'system1',
+    createdAt: '111',
+    updatedAt: '1111',
+    mapping: {
+      externalKey1: 'osmKey1',
+      externalKey2: 'osmKey2',
+    },
+  };
+
+  beforeEach(() => {
+    getSchemas = jest.fn().mockReturnValue([schema]);
+    getSchema = jest.fn().mockReturnValue(schema);
+    schemaManager = new SchemaManager({ getSchemas, getSchema });
+  });
+
+  afterEach(() => {
+    getSchemas.mockReset();
+  });
+
+  describe('getSchemas', () => {
+    it('return all schemas', async () => {
+      const res = await schemaManager.getSchemas();
+
+      expect(res).toBeInstanceOf(Array);
+    });
+  });
+
+  describe('getSchema', () => {
+    it('return specific schema', async () => {
+      const name = 'system1';
+      const expected = {
+        name: 'system1',
+        createdAt: '111',
+        updatedAt: '1111',
+        mapping: {
+          externalKey1: 'osmKey1',
+          externalKey2: 'osmKey2',
+        },
+      };
+
+      const res = await schemaManager.getSchema(name);
+
+      expect(res).toMatchObject(expected);
+    });
+    it('return undefined', async () => {
```

Comment:
this message is not informative, a more informative message can be

`it('should return undefined for non-existent schema')`

Reply:
Done